### DISCUSSION TITLE
[3.2.0.dev1] Minor Bugs and Usability Problems

### DISCUSSION THREAD

**Discussion by slook** [nicotine-plus/nicotine-plus#1535] [General]
Noteworthy minor but somewhat important changes that are needed to the existing components in future versions of the client

- Note: this is not a conversation thread, and as such, please reply in-line.
- Discussions in here do not require immediate reply nor action.

If you have an urgent query, please create a [New Issue in the Bug Tracker](https://github.com/nicotine-plus/nicotine-plus/issues/new/choose)

**Comment by slook** [comment]
Edit: **FIXED in Nicotine+ 3.2.0**

**The Setup Assistant cannot be closed from the first page, because no Close function is provided.**

Further more, no Escape binding is assigned. If it has been opened in error, the user may not wish to continue.

The only way to close it is by right-clicking on it in the Taskbar and closing it from there.

**Reply by mathiascode** [reply]
> Further more, no Escape binding is assigned. If it has been opened in error, the user may not wish to continue.

That's a system-specific bug.

**Reply by slook** [reply]
Oh really, what systems does it affect?

**Reply by slook** [reply]
On MX18 system its possible to Maximize it by double clicking its header, is that correct? It looks horrible.

And the context-menu does infact include a Close option (Alt+Space+C) but's it's not obivous.

**Reply by mathiascode** [reply]
> Oh really, what systems does it affect?

That I don't know, works on MX Linux 19.2 and my main system at least. Is it borked on MX Linux 18?

**Reply by slook** [reply]
Well, I wouldn't say it's borked, but it does have its own taskbar button says "Untitled window".

It can be maximized easily (in an effort to try and close out of it), a whole sea of grey on the MX18.3 install.

I know it doesn't do anything harmful. but thats only cos ive been thorugh it many times now (at first, I felt kinda forced to).

**Reply by mathiascode** [reply]
> Further more, no Escape binding is assigned.

Is this still an issue? I couldn't reproduce it.

**Reply by slook** [reply]
No, Escape is working now MX18, but the "Untitled window" taskbar item is still present:
![nicotine-plus Setup Assistant Untitled winow](https://user-images.githubusercontent.com/88614182/132130013-5c1a6bf1-c882-4591-a391-541274d5a3cc.png)

And there is no title bar as there is on MX19.
![nicotine-plus Setup Assistant no title bar](https://user-images.githubusercontent.com/88614182/132129522-d03c6805-61a3-416a-a306-5146f2d911e6.png)
_(Nicotine+ 3.2.0 Setup Assistant on MX18)_

May I suggest a "Manual Configuration..." button to close this window and open Preferences > Network, would make this window seem much less problematic.

The two buttons...
- "Setup Assistant..." _(continue)_
- "Manual Configuration..." _(close this and open Preferences > Network)_

**Reply by mathiascode** [reply]
> No, Escape is working now MX18, but the "Untitled window" taskbar item is still present:

Fixed. I also added a cancel button to the first page if all the the required steps have been completed previously.

**Comment by slook** [comment]
Edit: FIXED in Nicotine+ 3.2.0

**Private Chat cannot open menu on username click. Not left nor right mouse button.**

**Reply by slook** [reply]
Context-menu on chat tabs all menuitems text is all bold

**Reply by mathiascode** [reply]
> Context-menu on chat tabs all menuitems text is all bold

What system is this?

**Reply by slook** [reply]
MX18.3 XFCE 4.12 (but not MX19)

Infact, it is all the active secondary tabs, search, browse, info, chat.

_It doesn't bother me, just seemed strange I never noticed it before:_
![nicotine-plus Secondary tab contextmenu bold menuitems](https://user-images.githubusercontent.com/88614182/132133599-746f8b76-3ead-478c-ab35-4f683e62e035.png)

**Reply by slook** [reply]
But the bold text in the menu is no biggie, the main problems with Private Chat is username not clickable and https://github.com/nicotine-plus/nicotine-plus/issues/1512

**Reply by slook** [reply]
Private Chat username clickable support added https://github.com/nicotine-plus/nicotine-plus/commit/5fa91d69eb955e6e69b2b2d6378db7e22ad6a516 and https://github.com/nicotine-plus/nicotine-plus/commit/e49665f82fd0105262e4b566de62fb233fe1c3a5

**Comment by slook** [comment]
**Chat Text Entry box autocomplete list is unconstrained**

Too tall, so when it appears it covers the messages.

Suggest a height of 4 lines would be sufficient. Ideally so it fits no bugger than the bottom of the status bar.

**Reply by mathiascode** [reply]
I don't think this is possible. I suggest setting "Minimum characters required to display drop-down" to 3, if you're still using the old default.

**Reply by Nachtalb** [reply]
On that note I am sceptical about the usage of the feature itself. I think it's very unintuiitive as it is now. It opens the on every single word which is quite intrusive. (even when you have it set to 3 chars). 

IMHO the usual approach of typing `@`  opens the completion menu / tab completion. AFAIK this is the way every single chat app I know does it: `irssi`, `hexchat`, `the lounge`, `whatsapp` (tho I haven't used it for long I don't think it has changed), `telegram`, `github` and some more. 

It would also make the charachter threshold obosolete and the size question would also be a topic to talk about again buecause it is only opened on demand. 

A new option could be tho what prefix the people want. From IRC i know some people have set it to `.`  instead of `@` as it's faster to type.

**Reply by slook** [reply]
See related https://github.com/nicotine-plus/nicotine-plus/discussions/1535#discussioncomment-1455269

**Comment by slook** [comment]
Edit: FIXED in Nicotine+ 3.2.0

**`Ctrl+R` "Show Room List" does not work, so it should be removed from the Keyboard Shortcuts window.**

Unless, it is possible to have this shortcut jump to the Chat Rooms tab and open the Rooms List popover.

**Comment by slook** [comment]
Edit: FIXED in Nicotine+ 3.2.0

**Expand button icon does not show until it is clicked (Uploads) - specific to MX18**

![nicotine-plus Uploads expand button no icon](https://user-images.githubusercontent.com/88614182/132264537-8970832f-3891-4034-91ed-74a2e581fe4b.png)

Noticed since https://github.com/nicotine-plus/nicotine-plus/commit/288b8b86715c7312ba7be1814e72abaac4ff9af7 or https://github.com/nicotine-plus/nicotine-plus/commit/445118d8b28ac3e5bcfc2ce01005b9a96a151eae

**Reply by mathiascode** [reply]
https://github.com/nicotine-plus/nicotine-plus/commit/d3b6507c38f8bba2431b5111256bc74c8e6b3c0d

**Comment by slook** [comment]
_FIXED in Nicotine+ 3.2.1_ - **Chat message view scrolls up when resizing main window smaller**

...and then it stays there, freezing the view so that new messages are not seen, until manually scroll down.

**Reply by mathiascode** [reply]
Fixed in https://github.com/nicotine-plus/nicotine-plus/commit/77cc5645e97f13f30d18baa582958568d9f5e06f

**Reply by slook** [reply]
Reverted in https://github.com/nicotine-plus/nicotine-plus/commit/14ace1b00da222c0927485b31388a005e148bc48 for example the Room Wall went the bottom where this wasn't desired.

**Reply by slook** [reply]
This seems to be fixed now.

**Comment by slook** [comment]
_FIXED in Nicotine+ 3.2.0_ - **Incorrect MP3 bitrates reported by tinytag**

DONE: investigate and send bug report to https://github.com/devsnd/tinytag about wildly incorrect bitrate issue (issue raised by user in nicotine chatroom).

_many example CBR files reported as 96Kbps, 128Kbps, 144Kbps but is 320Kbps..._

_MP3 sometimes reported as about double 416 the actual bitrate of 224..._

_Edit: actually, a lot of files have incorrectly reported values. Most files are slightly off, and some are wildly off._

_Some VBR files report double 454 the actual average bitrate of 200ish..._

**Reply by mathiascode** [reply]
I submitted a fix for a few cases: https://github.com/devsnd/tinytag/pull/119

**Reply by slook** [reply]
https://github.com/nicotine-plus/nicotine-plus/commit/a73b86d108264b2390e592de0da9cbcd1a8d4767 and https://github.com/nicotine-plus/nicotine-plus/commit/529e68d4a203991cbdb866af3b4424618e425fa0 appears to remedy the issue for new or modified folders.

**Reply by mathiascode** [reply]
You need to re-add your share, remove the .db files or modify a file in the folder where you want to update metadata for other files, otherwise cached data is used when rescanning.

**Reply by mathiascode** [reply]
Keep in mind, this only fixes the invalid 320kbps files, not the VBR ones.

**Reply by slook** [reply]
Nice diagnoses and fix @mathiascode , much more correct looking values now... Search Results appear to be much more accurate, and there aren't so much odd ball bitrates (like 319Kbps) anymore it seems :-)
.

> fixes the invalid 320kbps files

And other bitrates seem more accurate too. I think this issue raised by the user has bought about a very important fix actually, upon inspection a large number of my own file tags were grossly misreported, it seems this has been a widespread issue for a long time so it is certainly worthy of a Changelog entry and Closed Bug report on the issue tracker.
.

> not the VBR ones

I have never known these to anywhere near accurate in any file listing application whatsoever (I believe there is some technical reason why it is so hard to calculate the average rate efficiently), so perhaps it is appropriate, at least for the time being, to state "VBR" in these cases?
.

> cached data is used when rescanning

How will the users easily recover from this bug without any sort of forced 'Rebuild All Shares' feature (that deletes the .db's and rebuilds from scratch such as is done to workaround issue https://github.com/nicotine-plus/nicotine-plus/issues/1519)? In regard to possibly adding such a function into the Shares menu, also suggest consider https://github.com/nicotine-plus/nicotine-plus/discussions/1526#discussioncomment-1283441 whilst doing so, and/or suggest consider closing https://github.com/nicotine-plus/nicotine-plus/issues/1514 to provide a means for end-users to achieve this purpose.

**Reply by slook** [reply]
- The issue you reported has now been fixed in Nicotine+ 3.2.0.dev1 (Unstable).
- Download latest version (all platforms) https://nicotine-plus.github.io/nicotine-plus/
> - Keep in mind, this only fixes the invalid CBR MP3 file bitrates, not the VBR ones.
> - You need to re-add your share, remove the .db files or modify a file in the folder where you want to update metadata for other files, otherwise cached data is used when rescanning.

[_laabla____] Thanks for the update, will go ahead and do that.
[_laabla____] Appears to have worked well! Thanks again.
[_oo_] CTCP VERSION
[_laabla____] Nicotine+ 3.2.0.dev1
[_oo_] Your feedback was most welcomed, thank you for raising the issue.
[_oo_] If you find any other bugs or have any other suggestions, please do speak out to one of the devs.
[_laabla____] Cheers, will do. Really appreciate the work on this client, it's excellent.

**Reply by mathiascode** [reply]
tinytag 1.7.0 contains a fix for cases where no audio info is returned for short MP3 files.

https://github.com/nicotine-plus/nicotine-plus/commit/93231c96579313fcfcc1ee55ee2fa63edd5fbd19 fixes the VBR issue for files with a sampling rate of >=32 khz. For anything below 32 khz, the info is still incorrect: https://github.com/devsnd/tinytag/issues/124

**Reply by mathiascode** [reply]
https://github.com/nicotine-plus/nicotine-plus/commit/dad61cc7e63660707afa28c8ae6c2879b1e650bc fixes the VBR issue for files with a sampling rate below 32 khz.

**Reply by mathiascode** [reply]
I think any issues related to mp3 files should be fixed now. Let me know if you still have a file that breaks things.

**Comment by slook** [comment]
Edit: FIXED in Nicotine+ 3.2.0

**Close via application window 'X' when 'Remember Choice: Quit Program' is set results in GTK error displayed on console**

It is interesting that `Menu > Quit` or `Ctrl+Q` always exits gracefully, but when using 'Close' from the OS window menu or pressing 'X' on the main window when 'Remember choice: Quit Program' option is active, then an error in the console during termination is always produced:

```
[2021-09-11 21:37:03] Disconnected from server 192.168.1.110:54920

(org.nicotine_plus.Nicotine:19500): Gtk-CRITICAL **: 21:37:03.866: gtk_range_get_adjustment: assertion 'GTK_IS_RANGE (range)' failed
user@machine:~/Git/nicotine-plus$ 
```

- If the 'Close Nicotine+?' confirmation dialog is displayed before exit, then the above error is somehow avoided.

Perhaps knowing the cause of the quirk investigated here will be useful in cleaning up the exit sequence when Nicotine+ is asked to close without using the Quit dialog, although please note I see no ill effect of this minor bug, which is reported for informational purposes only.

MX18.3 GTK 3.22.11 Nicotine+ 3.2.0.dev1, using Python 3.5.3
MX19.4 GTK 3.24.5 Nicotine+ 3.2.0.dev1, using Python 3.7.3

**Reply by mathiascode** [reply]
Fixed in https://github.com/nicotine-plus/nicotine-plus/commit/f509f1d2487ad497c4e5589f13e01785a9241aba

**Comment by slook** [comment]
Edit: FIXED in Nicotine+ 3.2.0

**Switching to new Chat Tab should always focus the Text Entry**

If there is more than 1 tab open then no widget is focused by default upon creation of a new private or room chat tab, nor when switching tabs. Text entry is not always possible until the user manually switches widget focus, the current behaviour is inconsistent.

- To solve this issue, the Text Entry should always have the focus upon chat tab activation.

Additionally, any previous scroll-up in any messages TextView or Activity View prevents new messages in an inactive tab from being visible after switching to it, which is severely compounded by the [bug that causes TextView scrolling to occur upon window resize](https://github.com/nicotine-plus/nicotine-plus/discussions/1535#discussioncomment-1294120).

- Ideally, forcefully scrolling to bottom of both TextView and Activity View upon each tab activation would greatly alleviate that issue also.

**Reply by Inso-m-niaC** [reply]
Focus stays on the last tab created currently (last private and last room). And if you type on any of the other tabs where the entry isn't focused and then press enter, the message is sent to chat of the last opened/created tab.

**Reply by mathiascode** [reply]
Fixed in https://github.com/nicotine-plus/nicotine-plus/commit/c891019be0846abd97928dc20ead3ee1b30de965 (turned out to be an annoying bug)

**Comment by slook** [comment]
DONE in Nicotine+ 3.2.0

**Preferences > Events > Transfer Actions > "Open Folder" string should be renamed to "Open in File Manager"**

Also, there should be a "Browse Folder" option, but this action does not exist in the dropdown listboxes, nor does "Retry" in the Upload option listbox. Those actions would seem to be the most logical, the fact they are missing seems like an oversight, but perhaps a separate default action is ideally required for Paused, Cancelled or Failed transfers. An expected (non-disabled) function is not provided [UI bug].

**Reply by mathiascode** [reply]
"Browse Folder" is a fairly recent feature, which I forgot to add to the double-click actions. Fixed in https://github.com/nicotine-plus/nicotine-plus/commit/549eb514783de8c1e6db0ac6684df2ac17bc54f7

**Comment by slook** [comment]
**Filter bar layout issues (Result Filters)** _(moved from Issue #1497)_

The Result Filter pane suffers from a quirky UI layout on narrow width viewports, due to the responsive stacking obstructed by the Free Slot, Clear and Help buttons. The behaviour is less than ideal, suggesting that overall UI enhancement(s) of the Result header / Result Filter pane layout are needed to better support narrow width viewport scenarios.

![nicotine-plus Result Filters layout issues](https://user-images.githubusercontent.com/88614182/136634924-944ccb2d-7d78-439b-855b-8acec6107fa7.png)
_Edit: Since https://github.com/nicotine-plus/nicotine-plus/commit/f436411d676f03432a704b7912f85d03526dec5c the layout has been somewhat improved now, although there are still issues at certain widths (pictured)._

I was thinking to move a couple items from the Filter bar, and put them in the header bar instead, something like this...

![nicotine-plus File Search Result Filters panel needs better responsive layout](https://user-images.githubusercontent.com/88614182/131263203-d93b259a-2e0a-49c8-ab9f-947a58ed0daa.png)


Anyway, I believe it makes sense to have these important frequently used widgets always accessible (the 'Clear all active filters' and 'Free slots') because they are odd-balls, and in any case I don't think they would do any harm if they were in the results header bar, it would saves on number of steps needed to clear filters too.

The target min width is 640 pixels or slightly less (because 1280 / 2 = half of a common screen resolution width). If it's borderline then consider rename the 'Result Filters [n]' button caption to 'Filters [n]'. At a main window width of 625px, a reasonable search phrase of up to 20 characters is possible, and prevents a lot of ugly layout issues in other parts of the application.

Keeping the Filter bar to just equally-sized filter text boxes only and nothing else would solve everything :-) If only things were that easy huh!

_DONE in 3.2.0 -_ If all else fails, remove the Help button, folks confuse it for the main application help button anyway, and then the complex info about RegEx scares people! The bubble is too long and confusing... Help could perhaps be better accessed by a context menu of any Filter related object instead, or simply put the better info into the ToolTips and scrap the Help bubble. Advanced Filter Help is available from the Preferences so that's good enough imo... Definitely do not put the button in the results header nor main window.

**Reply by mathiascode** [reply]
Improved this slightly in the latest commits.

**Reply by slook** [reply]
That's nice the separator works well.

**Comment by slook** [comment]
_FIXED in Nicotine 3.2.0 -_ **Focus is grabbed away from any widget in the application upon a new (Wish) tab being created in Search Files**

There is a slight issue I have noticed the focus is grabbed away from any widget in the application upon a new (Wish) tab being created, regardless of which main window tab is active. This is most noticeable when typing into a chat text entry for example.

_Edit: This minor UI bug was not caused by this commit._

_I think this might be caused by on_switch_search_page() in search.py:_
https://github.com/nicotine-plus/nicotine-plus/blob/491cd0a732e83c1b5fabc53f98b7750530b078e3/pynicotine/gtkgui/search.py#L93-L94 _It doesn't check if the main window tab is active or not, only tab.Main (search tabs)._

**Reply by mathiascode** [reply]
Fixed in https://github.com/nicotine-plus/nicotine-plus/commit/82fa35affadb20bebc6663a9628f3413c98f0ab9

**Comment by 82d** [comment]
there is a problem when you enter letters into the search bar when you find it in the history after the second match, it is automatically added. 
the problem is not even that the characters are added when they are not asked for, but that after that they are automatically highlighted. this fact knocks down what is found in the additional clipboard, which is captured by the middle mouse wheel. and often when using a double clipboard, this file prevents this from happening.
I think the solution will be such as not to supplement the line automatically, it is absolutely meaningless having a history of only 15 lines. you can always use the arrows to select when needed.

**Reply by slook** [reply]
it seems like the tab completion fills the text entry beyond the 'Minimum character required to display drop-down' setting, so I suppose these two features somewhat interfere with each other. See related https://github.com/nicotine-plus/nicotine-plus/discussions/1535#discussioncomment-1281216

I don't think the text entry that have thier own history combo drop-down should have the text completion drop-downs enabled, because they look so similar this is confusing.


> it is absolutely meaningless having a history of only 15 lines

_DONE -_ Agreed, this recently added feature would be much more useful if this was bigger/configuarable , see https://github.com/nicotine-plus/nicotine-plus/discussions/1526#discussioncomment-1280193

> after that they are automatically highlighted

Refining/disabling the erratic text entry tab completion behavior would close #1607

**Reply by slook** [reply]
https://github.com/nicotine-plus/nicotine-plus/commit/e1818ab91d5a28d94735dd5f9b409a2e67a49bf5 and https://github.com/nicotine-plus/nicotine-plus/commit/9268dab82c65924ce1143c989a4436d588d2cf58 has greatly improved the usability of the text entry.

**Reply by 82d** [reply]
> ihas greatly improved the usability of the text entry.
> 
auto-completion has not been removed. while typing, it often gets in the way, it seems now it has begun to add characters even faster than it was. it was more convenient if you did it as it is now implemented in Wishlist.

here is a simple example. enter a multi-word phrase in the search.
this is now in the history; then imagine that you need to transfer the same combination but additionally by copying paste via button 2
ctrl+v / button 2
due to automatic phrase selection, the button 2 buffer is deleted.

**Reply by slook** [reply]
I am able to paste with keyboard shortcut from the clipboard half way through a completion with no problem, nothing is lost. Obviously if pasting at the end is wanted this needs the pressing of Right arrow key first, and this works as expected too.

Pressing the middle mouse button to paste does not work after the first time, yet the clipboard data still exists. I don't know why completion effects it but it does seem to (Edit: this happens after a manual selection as well). I never noticed before because if I was already typing then I use the keyboard to paste so it doesn't cause a major issue.

Even after deleting the text entry pasting with middle button is not possible until the data is copied again from the source, but this happens wether or not a completion event has happened or not, so I don't think that problem is related to completion.

N+ does not handle any paste operations, I think middle mouse button paste comes from the OS?

Update: This bug might belong to Gtk and could be specific to X11 or Xfce, currently investigating.

**Reply by 82d** [reply]
>Even after deleting the text entry pasting with middle button is not possible until the data is copied again from the source, but this happens wether or not a completion event has happened or not, so I don't think that problem is related to completion.

the problem is not because of the completion itself (although I consider it useless), the buffer is lost due to the automatic selection of the completed piece of text. 
I think it will be easier to abandon auto-completion altogether or make it possible to disable it optionally. auto-completion is a hindrance when I re-write some text but not a full copy, then this auto-completion needs to be deleted. it is much better to choose matches with arrows yourself. I really want it to be possible to turn it off, this will solve two problems at once.

**Reply by mathiascode** [reply]
An option would make more sense. Completely removing auto-completion would just result in more issue reports and feature requests about missing auto-completion.

**Reply by 82d** [reply]
if we take for example a keyboard in a smartphone, I think there will be an option in any one so that I can turn off the auto-completion function. so that I can continue working with her comfortably. why not do the same in Nicotine+. especially taking into account the fact that this harms the copy-paste search functionality in full, which is used directly to the destination. 
 I consider the experience of virtual keyboards to be significant in order to take this option into account.:)
 I suggest adding a checkbox to the menu or if you don't like it then then go to the "Searches" settings
![Auto-comp](https://user-images.githubusercontent.com/15121782/137648294-a975bea1-4371-49f6-ad07-2fb06c3ac01f.png)
.

**Reply by 82d** [reply]
or maybe such a solution to this problem. 
The selection is used to quickly remove auto-additions to the Backspace.
do the same thing, but without highlighting. that is, the add-on can be deleted in the same way, but without highlighting it, delete it to the place from which it started.

**Comment by slook** [comment]
_RESOLVED in Nicotine 3.2.0_ - **'Select previous tab on startup' only remembers main window mode**

The name and location of this option in User Interface > Tabs implies that the child tab will be remembered, but it isn't. Either...

- _DONE_ - **The option should be relocated into the 'Visible Modes' and renamed to 'Remember previous mode on startup'**, or;
- The function could be expanded to remember previous child tab if possible (for which chat tab was last active).

**Reply by 82d** [reply]
> **'Select previous tab on startup' only remembers main window mode**
> 
> The name and location of this option in User Interface > Tabs implies that the child tab will be remembered, but it isn't. Either...
> 
>     * The option should be relocated into the 'Visible Modes' and renamed to 'Remember previous mode on startup', or;
> 
>     * The function could be expanded to remember previous child tab if possible (for which chat tab was last active).

I want to add that it would not be bad to also add an option to save tabs with search results. this would be an advantage for many against SoulseekQt.
the reason why it is convenient is simple. you don't have to worry about saving search results every time you exit. when you need to exit abruptly or an emergency shutdown of the computer. search results are sometimes very valuable. this will be another advantage of nicotine.

**Reply by slook** [reply]
We have already seen your suggestion about saving search results in https://github.com/nicotine-plus/nicotine-plus/issues/1437

I suggest it would be best to re label the option discussed here to 'Remember previous mode on startup' since such functionality is not possible at the moment.

**Reply by mathiascode** [reply]
I've separated the 'Tabs' section into 'Primary Tabs' and 'Secondary Tabs' now (https://github.com/nicotine-plus/nicotine-plus/commit/2bc4439da844ebc0fc5117cf855ab4fe68ab09e5).

**Comment by 82d** [comment]
I noticed when working with the search not using the keyboard but with one mouse. it would be convenient if there was a button to clear the search bar. the same as is done on the User Browse pages. it appears with the beginning of printing.


![Снимок экрана_2021-10-16_06-11-17](https://user-images.githubusercontent.com/15121782/137647156-9fc32cf8-c3b1-4b40-a7b6-42193f4fdbdb.png)

**Comment by slook** [comment]
_RESOLVED in Nicotine 3.2.0_ - **Taskbar notification won't clear after receiving notification of mention in public chat whilst minimized**

Even with the 'Show notification popup when you are mentioned in a chat room' option disabled, the Xfce window manager's taskbar button continuously blinks until the application is restarted. There doesn't seem to be any other way of clearing this.

**Reply by mathiascode** [reply]
I can't reproduce this for some reason. Urgency hints should always be cleared when the window is focused:
https://github.com/nicotine-plus/nicotine-plus/blob/c9354fe0f41a49e9804b514bb46ceefd4f279a3f/pynicotine/gtkgui/frame.py#L295-L304

**Reply by slook** [reply]
on test debug log line 296 in `on_window_active_changed()` is never called.

_FIXED in Nicotine 3.2.0_
